perf/perf_datatype_profiling.py

```diff
@@ -0,0 +1,199 @@
+            base_packages += [
+            	"kernel-default-debuginfo",
+		        "kernel-debug-debuginfo",
+		        "kernel-default-devel",
+		        "kernel-devel",
+		        "kernel-default"
+            ]
+        else:
+            self.cancel("Unsupported Linux distribution")
+
+        # Try versioned first, then fallback to unversioned
+        missing = []
+        for pkg in base_packages:
+            versioned_pkg = f"{pkg}-{kernel_version}.ppc64le"
+            unversioned_found = False
+
+            # Try versioned package
+            if self.run_cmd(f"rpm -q {versioned_pkg}").strip().endswith("is not installed"):
+                # Try unversioned
+                if self.run_cmd(f"rpm -q {pkg}").strip().endswith("is not installed"):
+                    missing.append(pkg)
+
+        if missing:
+            self.cancel(f"Missing required packages: {', '.join(missing)}")
+
+
+    def check_perf_report_headers(self, cmd):
+        """
+        Checks for expected headers in perf report output.
+        """
+        output = self.run_cmd(cmd)
+        expected_headers = ["Symbol", "Data Type", "Data Type Offset"]
+        for header in expected_headers:
+            if header not in output:
+                self.fail(f"Missing expected header '{header}' in perf report output")
+
+
+    def check_perf_annotate_headers(self, cmd, check_insn_stat=False):
+        """
+        Checks for expected headers in perf annotate output.
+        """
+        self.run_cmd(f"{cmd} > out")
+        output = self.run_cmd("cat out")
+
+        if check_insn_stat:
+            # First check for Name/opcode, Good, Bad headers
+            expected_insn_headers = ["Name/opcode", "Good", "Bad"]
+            for header in expected_insn_headers:
+                if header not in output:
+                    self.fail(f"Missing expected header '{header}' in perf annotate insn stat output")
+
+        # Now check for 'offset' and 'size' in each Annotate type block
+        sections = output.split("Annotate type:")
+        for section in sections[1:]:
+            if "offset" not in section or "size" not in section:
+                self.fail("Missing 'offset' or 'size' header in perf annotate output")
+
+
+    def test_datatype_profiling(self):
+        """
+        Test to verify perf data type profiling feature.
+        Steps:
+        1. Verify perf report headers.
+        2. Verify perf annotate data type headers.
+        3. Verify perf annotate instruction stats headers.
+        Repeat above for:
+            a. perf mem record -a sleep 10
+            b. perf record -a -e mem-loads sleep 5
+            c. perf record -c 1 -e mem-stores sleep 5
+        """
+
+        self.check_dependencies()
+
+        record_cmds = [
+            "perf mem record -a sleep 10 > /tmp/perf_output.data",
+            "perf record -a -e mem-loads sleep 5 > /tmp/perf_output.data",
+            "perf record -c 1 -e mem-stores sleep 5 > /tmp/perf_output.data"
```

Comment:
you can use -o with record and -i with report options instead of redirecting output to a file